backend/controllers/bulkAppointmentsUpload.js

```diff
@@ -0,0 +1,86 @@
+const bulkAppointmentsUploadFactory = (csvParserService, offenderLoader, logError) => {
+  const index = async (req, res) => {
+    try {
+      if (!req.session.data) {
+        res.render('error.njk', {
+          url: '/need-to-upload-file',
+        })
+        return
+      }
+
+      res.render('bulkUploadFile.njk', {
+        title: 'Upload a CSV File',
+        errors: req.flash('errors'),
+        appointmentDetails: req.session.data,
+      })
+    } catch (error) {
+      logError(req.originalUrl, error, 'Sorry, the service is unavailable')
+      res.render('error.njk', {
+        url: '/need-to-upload-file',
+      })
+    }
+  }
+
+  const getNonExistingOffenderNumbers = (uploadedList, prisonersList) => {
+    const matchingPrisonerNumbers = []
```

Comment:
This is a bit inconsistent, could do with converting this to be a couple of maps maybe? Can pair up on it if you like.